plugin/audit_log/filter.c

```diff
@@ -0,0 +1,242 @@
+/* Copyright (c) 2016 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation; version 2 of
+   the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA */
+
+#include <my_global.h>
+#include <my_sys.h>
+#include <my_user.h>
+#include <mysql_com.h>
+#include <hash.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "filter.h"
+
+typedef struct
+{
+  char name[USERNAME_LENGTH + HOSTNAME_LENGTH];
+  size_t length;
+} account;
+
+static HASH include_accounts;
+static HASH exclude_accounts;
+
+#if defined(HAVE_PSI_INTERFACE)
+
+static PSI_rwlock_key key_LOCK_account_list;
+static PSI_rwlock_info all_rwlock_list[]=
+{{ &key_LOCK_account_list, "audit_log_filter::account_list", PSI_FLAG_GLOBAL}};
```

Comment:
Is the "::" convention used somewhere else for key names like that? I thought it was "_" everywhere